pkg/certificate/certificate.go

```
@@ -0,0 +1,29 @@
+package certificate
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+func GetCert(c corev1listers.SecretLister, name, namespace string) (*x509.Certificate, *rsa.PrivateKey, error) {
+	caSecret, err := c.Secrets(namespace).Get(name)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	block1, _ := pem.Decode(caSecret.Data["tls.crt"])
```

we need handle the case that the secret does not have the `tls.crt`

---

and don't swallow the error